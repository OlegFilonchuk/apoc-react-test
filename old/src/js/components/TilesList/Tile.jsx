import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Tile.less';

const MAX_VISIBLE_FOOTER_ITEMS = 3;

const formProtocolsListString = list => {
  const listWithUpToThreeDescItems = list.slice(0, MAX_VISIBLE_FOOTER_ITEMS);
  const concatenatedDescItems = listWithUpToThreeDescItems.join(', ');
  const shouldStringHavePeriods = list.length > MAX_VISIBLE_FOOTER_ITEMS;

  const lineEnding = shouldStringHavePeriods ? ' ...' : '';

  return concatenatedDescItems + lineEnding;
};

export default function Tile({ title, link, amount, footerItems, onClick, className }) {
  const tileClass = 'tile';
  const tileContainerClass = classNames(`${tileClass}__container`, className);
  const tileMainClass = `${tileClass}__main`;
  const tileNameClass = `${tileMainClass}__name`;
  const tileMainAmountClass = `${tileClass}__main__amount`;
  const tileFooterClass = `${tileClass}__footer`;

  const onTileClick = event => {
    if (typeof link === 'string' && link.length) {
      window.location.href = link;
    }

    onClick(event);
  };

  const TileAmount = amount !== null ? <div className={tileMainAmountClass}>{amount}</div> : null;

  const TileFooterString = footerItems.length ? formProtocolsListString(footerItems) : null;

  return (
    <div className={tileContainerClass}>
      <div className={tileClass} onClick={onTileClick}>
        <div className={tileMainClass}>
          <div className={tileNameClass}>{title}</div>
          {TileAmount}
        </div>

        <div className={tileFooterClass}>{TileFooterString}</div>
      </div>
    </div>
  );
}

Tile.propTypes = {
  /**
   * Tile name
   */
  title: PropTypes.string.isRequired,

  /**
   * Link for onClick function handler.
   */
  link: PropTypes.string.isRequired,

  /**
   * Amount of items that are nested under this tile.
   */
  amount: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),

  /**
   * List of nested protocols for this Tile
   */
  footerItems: PropTypes.arrayOf(PropTypes.string),

  /**
   * onClick callback function
   */
  onClick: PropTypes.func,

  /**
   * Any custom class names
   */
  className: PropTypes.string
};

Tile.defaultProps = {
  footerItems: [],
  onClick: () => true,
  amount: null,
  className: ''
};
