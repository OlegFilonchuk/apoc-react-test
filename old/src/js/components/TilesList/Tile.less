@import 'TilesVariables';

.tile__container {
  float: left;
  width: @tile-container-width;
  overflow: hidden;

  /* all after 3rd */
  &:nth-child(n + 4) {
    margin-top: @margin-between-rows-in-tiles-list;
  }

  /* 1st, 4th, 7th - each first tile in a row */
  &:nth-child(3n + 1) {
    .tile {
      float: left;
    }
  }

  /* 3rd, 6th, 9th - each last tile in a row */
  &:nth-child(3n) {
    .tile {
      float: right;
    }
  }

  /* 2nd, 5th, 8th - each central tile in a row */
  &:nth-child(3n - 1) {
    .tile {
      margin: 0 auto;
    }
  }

  .tile {
    font-family: open-sans-regular;
    border: 1px solid @tile-main-blue-color;
    color: @tile-main-blue-color;
    width: @tile-width;
    height: @tile-full-height;
    display: flex;
    flex-direction: column;
    cursor: pointer;

    &:hover {
      color: @tile-hovered-blue-color;
      border-color: @tile-hovered-blue-color;

      .tile__footer {
        border-color: @tile-hovered-blue-color;
        background-color: @tile-hovered-blue-color;
        color: @tile-footer-hovered-text-color;
      }
    }

    &__main {
      flex: 1;
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;

      &__name {
        font-size: @tile-title-font-size;
      }

      &__amount {
        font-size: @tile-amount-font-size;
      }
    }

    &__footer {
      background-color: @tile-main-blue-color;
      color: @tile-footer-text-color;
      min-height: @tile-footer-min-height;
      text-overflow: ellipsis;
      white-space: nowrap;
      overflow: hidden;
      text-align: center;
      line-height: @tile-footer-height;
      padding: 0 @tile-footer-side-padding;
      font-size: @tile-amount-font-size;
    }
  }
}
