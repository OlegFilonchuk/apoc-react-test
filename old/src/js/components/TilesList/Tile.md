## Tile is a single item for TilesList

Complex component

```
const protocolsList = [
    'Adobe',
    'Skype',
    'IMAP',
    'Jabber'
  ];
  
(
  <div style={{ overflow: "hidden" }}>
    <Tile
      title="Imported Scenarious"
      link="#Tile"
      amount={protocolsList.length}
      footerItems={protocolsList}
    />
  </div>
)
```

Simple component

```
(
  <div style={{ overflow: "hidden" }}>
    <Tile
      title="Imported Scenarious"
      link="#TilesList"
    />
  </div>
)
```
