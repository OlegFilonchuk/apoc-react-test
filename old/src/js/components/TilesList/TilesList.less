@import 'TilesVariables';

.tiles-list {
  min-width: @tiles-list-min-width;
  overflow: hidden;
}
