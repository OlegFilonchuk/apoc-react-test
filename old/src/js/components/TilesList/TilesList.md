## TilesList

```
  const protocolsList = [
    'Adobe',
    'Sharepoint',
    'MySQL',
    'Skype',
    'IMAP',
    'Jabber'
  ];
  
  const shortList = [
    'MySQL',
    'Skype'
  ];

  (
  <TilesList>
    <Tile
      title="First Pack"
      link="#testLink"
      amount={protocolsList.length}
      footerItems={protocolsList}
    />
    
    <Tile
      title="Second Pack"
      link="#testLink"
      amount={shortList.length}
      footerItems={shortList}
    />
    
    <Tile
      title="Another item"
      link="#testLink"
      amount={protocolsList.length}
      footerItems={protocolsList}
    />
    
    <Tile
      title="Nex Row Pack"
      link="#testLink"
      amount={protocolsList.length}
    />

    <Tile
      title="No Protocols Pack"
      link="#testLink"
    />
  </TilesList>
  )
```
