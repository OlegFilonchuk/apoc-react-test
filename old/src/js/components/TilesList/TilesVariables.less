@tile-main-blue-color: #0290d7;
@tile-footer-text-color: #ffffff;

@tile-hovered-blue-color: #005da4;
@tile-footer-hovered-text-color: #ffffff;

@tile-width: 235px;
@tile-full-height: 85px;

@tile-footer-height: 27px;
@tile-footer-min-height: 10px;
@tile-footer-side-padding: 10px;

@tile-title-font-size: 16px;
@tile-amount-font-size: 12px;

@margin-between-rows-in-tiles-list: 20px;
@number-of-columns-in-tiles-list: 3;

@tiles-list-min-width: calc(@tile-width * @number-of-columns-in-tiles-list); // as there grid of 3 tiles in a row
@tile-container-width: calc(100% / @number-of-columns-in-tiles-list);
