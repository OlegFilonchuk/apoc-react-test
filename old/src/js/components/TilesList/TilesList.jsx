import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './TilesList.less';

export default function TilesList({ children, className }) {
  const finalClassName = classNames('tiles-list', className);

  return <div className={finalClassName}>{children}</div>;
}

TilesList.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.arrayOf(PropTypes.element)]),

  /**
   * Any custom class names
   */
  className: PropTypes.string
};

TilesList.defaultProps = {
  children: [],
  className: ''
};
