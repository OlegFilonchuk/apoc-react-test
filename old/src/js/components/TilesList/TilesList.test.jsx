import React from 'react';
import { shallow } from 'enzyme';

import Tile from './Tile';
import TilesList from './TilesList';

describe('<TilesList />', () => {
  describe('when rendering', () => {
    it('should have correct html', () => {
      const protocols = ['One', 'Two', 'Three'];

      const tilesList = shallow(
        <TilesList>
          <Tile title="Test title" link="#testLink" amount={protocols.length} footerItems={protocols} />
        </TilesList>
      );

      expect(tilesList).toMatchSnapshot();
    });
  });
});
