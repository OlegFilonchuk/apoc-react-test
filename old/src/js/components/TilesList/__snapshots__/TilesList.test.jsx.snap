// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TilesList /> when rendering should have correct html 1`] = `
<div
  className="tiles-list"
>
  <Tile
    amount={3}
    className=""
    footerItems={
      Array [
        "One",
        "Two",
        "Three",
      ]
    }
    link="#testLink"
    onClick={[Function]}
    title="Test title"
  />
</div>
`;
